### ISSUE TITLE
Various Google Apps stop working with microG (YouTube, Maps, Google Opinion Rewards, etc.)

### ISSUE THREAD

**Issue by phucvu4324** [microg/GmsCore#2680]
**Affected app**
Name: Google Opinion Rewards, Google App, Gmail, Maps, YouTube 
Package id 
`com.google.android.apps.paidtasks`
`com.google.android.gm`
`com.google.android.googlequicksearchbox`
`com.google.android.apps.maps`
`com.google.android.youtube`

**Describe the bug**
Google Apps including Opinion Rewards, Google App, Gmail, Maps, YouTube stopped working since last app update

**To Reproduce**
1. Install/Update Opinion Rewards with versionCode `2019049870`, Google App with versionCode `301437738`, Gmail with versionCode `64884528`, Maps with versionCode `1067838727`, YouTube with versionCode `1550046656`
2. Open affected apps
4. See error. It says that Play Services isn't installed or exit itself if installed without old previous app data, Account login function also does not work at all

**Expected behavior**
Apps should works with account, does not show any dialogues show that it doesn't have Play Services installed.
If i downgrade these affected apps to older version, it working again.

**Screenshots**
![Image](https://github.com/user-attachments/assets/09036890-8c20-4cd2-94bd-a0d86b6c9782)
![Image](https://github.com/user-attachments/assets/53489ab2-6363-4d33-b1a9-ea0613ebc52d)
![Image](https://github.com/user-attachments/assets/2b96b852-c59a-4971-b60f-c8e90759a810)
![Image](https://github.com/user-attachments/assets/9e019832-461e-40c7-b1e1-460a9b70731f)
![Image](https://github.com/user-attachments/assets/c6c8e375-9692-4cb0-80af-c635ef5da711)

**System**
Android Version: 14
Custom ROM: LineageOS 21 : `21-20241101-microG-akari`

**microG**
microG Core version: 0.3.4.240913
microG Self-Check results: All ticked

![Image](https://github.com/user-attachments/assets/b36cfc59-fcba-4512-bb48-f7dfc68e10ac)
![Image](https://github.com/user-attachments/assets/dd28bcd8-758c-4f22-a741-7a5e44df9419)
![Image](https://github.com/user-attachments/assets/4108d925-b6cc-4118-90ea-030f2be0dfda)

**Additional context**
This might also affects to more Google apps in future

**Comment by phucvu4324** [comment]
I tried with 0.3.5.240913, but its still the same, nothing changed.

**Comment by mcgi5sr2** [comment]
0.3.2.240913-21 same issue.

I think their is another issue open that covers this, so potential duplicate

**Comment by Quills245** [comment]
Family link also seems to be broken after updating to 2.42.0.T.702836465

Downgrading to Family Link 2.41.0.S.703253343 worked

**Comment by phucvu4324** [comment]
> 0.3.2.240913-21 same issue.
> 
> I think their is another issue open that covers this, so potential duplicate

it affects any microG version, as far as i tested from 0.2.26 till 0.3.5

**Comment by phucvu4324** [comment]
> I think their is another issue open that covers this, so potential duplicate
i don't think so, there just issue about Maps doesn't work

**Comment by LZeugirdor** [comment]
Can confirm this is happening for me on Calyx OS on the Pixel 4A. Gmail JUST got updated and is no longer working, my google voice has been out all day today and yesterday.

**Comment by Tentos** [comment]
I have installed `com.google.android.googlequicksearchbox` and tried to capture possible errors with [LogFox](https://f-droid.org/de/packages/com.f0x1d.logfox/).

I have stumbled upon the following messages:
```
1733945761.615 10281 24955 24983 W GoogleApiManager: The service for ctpu is not available: ConnectionResult{statusCode=SERVICE_INVALID, resolution=null, message=null}
1733945761.615 10281 24955 25984 W GooglePlayServicesUtil: com.google.android.googlequicksearchbox requires Google Play services, but their signature is invalid.
1733945761.615 10281 24955 25984 E GooglePlayServicesUtil: GooglePlayServices not available due to error 9
```
Unfortunately, I cannot interpret the meaning of "signature is invalid" - maybe the spoofed microg signature is no longer considered up to date?

I can upload the entire log file if necessary (I will however have to make sure that no sensitive information is included).

Device: OnePlus 6, /e/OS T 2.5; microg version 0.3.2.240913-102

**Comment by stanwood77** [comment]
Confirmed. Same issue on /e/OS Version 2.5T stable (Android 13) - Fairphone 4 not rooted.
Google calendar + Google mail stopped working last night further both app updates.

**Comment by gasimlyrashad** [comment]
I can confirm, Google Drive can't find my Google account. I will give the logs here.

**Comment by personofnorank** [comment]
Confirmed issue on e/OS 2.5, Fairphone 5 unrooted, MicroG 0.3.2.240913-102

**Comment by tjanez** [comment]
Here are the versions I needed to downgrade to in order for the Google Apps to start working again with microG 0.3.2.240913-21 (9f758d8) that comes with CalyxOS 6.2.0:

- Google Drive:
  - Version: 2.24.467.5.all
  - Version code: 213814724
  - https://www.apkmirror.com/apk/google-inc/drive/drive-2-24-467-5-all-release/
- Google Maps:
  - Version: 24.47.05.699977727
  - Version code: 1067804539
  - https://www.apkmirror.com/apk/google-inc/maps/maps-24-47-05-699977727-release/
- Google Chat:
  - Version: 2024.11.17.700122806.Release
  - Version code: 5580568
  - https://www.apkmirror.com/apk/google-inc/hangouts-chat/hangouts-chat-2024-11-17-700122806-release-release/

The steps for downgrading:
1. Uninstall the app (CAUTION: Your app permissions and settings will be lost!).
2. Find the app in Aurora Store and click on the three dots in the top right corner and select "Manual download".
3. Enter the desired Version code and click "Check". The app will be downloaded and installed afterwards.

**Comment by mar-v-in** [comment]
Google seemingly has updated their signature checking code. On Android 9 and above, the latest code uses [`PackageInfo.signingInfo`](https://developer.android.com/reference/android/content/pm/PackageInfo#signingInfo) (via [`GET_SINGNING_CERTIFICATES` flags](https://developer.android.com/reference/android/content/pm/PackageManager#GET_SIGNING_CERTIFICATES)) instead of [`PackageInfo.signatures`](https://developer.android.com/reference/android/content/pm/PackageInfo#signatures) (via [`GET_SIGNATURES` flag](https://developer.android.com/reference/android/content/pm/PackageManager#GET_SIGNATURES)). This will likely roll out to non-Google apps in the future as well.

This means that **all signature spoofing patches need to be updated** to address the change of API usage.

**Comment by zykure** [comment]
@tjanez 
Thanks, can confirm this solves the issue for now! I also had to downgrade Calendar:

* Google Calendar:
   * Version: 2024.46.2-700284849
   * Version code: 2017604273
   * https://www.apkmirror.com/apk/google-inc/calendar/calendar-2024-46-2-700284849-release-release/

**Comment by ale5000-git** [comment]
Since plain LineageOS does support signature spoofing, did anyone submit this issue to them?

**Comment by mar-v-in** [comment]
Someone already submitted the patch from @jonathanklee to LineageOS: https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/411374

**Comment by pnaf** [comment]
Downstream projects such as /e/OS have begun applying the patch: https://gitlab.e.foundation/e/os/android_frameworks_base/-/commit/2b9c74a4409f41924905c4b28aa900904e442992

I also saw this announcement from a few of /e/OS's social media accounts:

> A fix is available and will be deployed through an OTA update early next week.

**Comment by phucvu4324** [comment]
> Someone already submitted the patch from [@jonathanklee](https://github.com/jonathanklee) to LineageOS: https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/411374

is it can be included on next microG Service Core update? I wonder if lineage-microG also apply this patch

**Comment by ale5000-git** [comment]
> is it can be included on next microG Service Core update? I wonder if lineage-microG also apply this patch

"microG Service Core" is an app, the patch is for the ROM so it is completely unrelated.

The patch for LineageOS isn't yet merged, once merged it will likely picked up by every other ROM.

**Comment by phucvu4324** [comment]
> > is it can be included on next microG Service Core update? I wonder if lineage-microG also apply this patch
> 
> "microG Service Core" is an app, the patch is for the ROM so it is completely unrelated.
> 
> The patch for LineageOS isn't yet merged, once merged it will likely be picked up by every other ROM.

what for `-hw` variant? i think `fake-signature` also must be updated with new method

**Comment by ale5000-git** [comment]
On the -hw version is Huawei that handle the signature spoofing, so only Huawei can change it.
But are you sure that Huawei is also affected by this issue?

**Comment by petefoth** [comment]
>  I wonder if lineage-microG also apply this patch

As @ale5000-git says
> The patch for LineageOS isn't yet merged, once merged it will likely be picked up by every other ROM.

The Lineageos for microG project will monitor the progress of the patch in the LineageOS gerrit. We will only act if LineageOS decide ***not*** to merge the patch

**Comment by heinhuiz** [comment]
Speech Recognition & Synthesis (com.google.android.tts) is also affected.
- Last working version: googletts.google-speech-apk_20241030.02_p0.691594814
- Version number: 210521474

**Comment by dimka40185** [comment]
> On the **-hw** version is Huawei that handle the signature spoofing, so only Huawei can change it.
> 
> But are you sure that Huawei is also affected by this issue? Since the way signature spoofing is handled is different the problem may be different or may not happen.

Huawei has no problems, everything works as it worked. No problems

**Comment by mar-v-in** [comment]
> Could the self-check code also be updated to account for this?

Yes it will be. :)

**Comment by ale5000-git** [comment]
@mar-v-in
I haven't tried but looking at the code doesn't it always fallback to info.signature?
If I'm not wrong it still pass if info.signingInfo is null and info.signatures is not null even is useSigningInfo is true.

Shouldn't there be an useSigningInfoONLY (if Android >=9)?

**Comment by ale5000-git** [comment]
The patch for LineageOS was updated:
```patch
diff --git a/services/core/java/com/android/server/pm/ComputerEngine.java b/services/core/java/com/android/server/pm/ComputerEngine.java
index 275746f..264665d 100644
--- a/services/core/java/com/android/server/pm/ComputerEngine.java
+++ b/services/core/java/com/android/server/pm/ComputerEngine.java
@@ -168,6 +168,7 @@
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
@@ -1588,6 +1589,18 @@
 
             generateFakeSignature(p).ifPresent(fakeSignature -> {
                 packageInfo.signatures = new Signature[]{fakeSignature};
+                try {
+                    packageInfo.signingInfo = new SigningInfo(
+                            new SigningDetails(
+                                    packageInfo.signatures,
+                                    SigningDetails.SignatureSchemeVersion.SIGNING_BLOCK_V3,
+                                    SigningDetails.toSigningKeys(packageInfo.signatures),
+                                    null
+                            )
+                    );
+                } catch (CertificateException e) {
+                    Slog.e(TAG, "Caught an exception when creating signing keys: ", e);
+                }
             });
 
             return packageInfo;
```

**Comment by mar-v-in** [comment]
@ale5000-git `signingInfo` should never be null on SDK >= 28 if `GET_SIGNING_CERTIFICATES` is set. Thus we should never be able to enter the `signatures` section if `useSigningInfo` is set - except when an app has multiple signers as it does in Google's checking code.

**Comment by ale5000-git** [comment]
> [@ale5000-git](https://github.com/ale5000-git) `signingInfo` should never be null on SDK >= 28 if `GET_SIGNING_CERTIFICATES` is set. Thus we should never be able to enter the `signatures` section if `useSigningInfo` is set - except when an app has multiple signers as it does in Google's checking code.

On SDK >= 28 it should never be null for normal apps, but it can be null for spoofed apps with an old signature spoofing patch.
This is what we testing, or I'm wrong?

**Comment by mar-v-in** [comment]
No, with an old signature spoofing patch it will be the original app signature (in this case microG's signature), not null.

**Comment by ale5000-git** [comment]
> No, with an old signature spoofing patch it will be the original app signature (in this case microG's signature), not null.

And with the new patch if it thrown CertificateException what would be in packageInfo.signingInfo?

**Comment by mar-v-in** [comment]
It should never throw `CertificateException`, that's only when the certificate is invalid - which defeats the purpose of using signature spoofing (because you would be trying to spoof an invalid certificate, which nobody can use). In that case, `signingInfo` would be unchanged, that means it has the original app certificate (again microG's certificate in our case)

**Comment by ale5000-git** [comment]
> It should never throw `CertificateException`, that's only when the certificate is invalid - which defeats the purpose of using signature spoofing (because you would be trying to spoof an invalid certificate, which nobody can use). In that case, `signingInfo` would be unchanged, that means it has the original app certificate (again microG's certificate in our case)

In this Xposed module for signature spoofing if something go wrong signingInfo can be null (if I have undestrood the code correctly): https://github.com/whew-inc/FakeGApps/pull/23/files

**Comment by mmvanheusden** [comment]
Google Photos seems to be broken as well. (version 7.10.0.704893900).
![Google Photos](https://github.com/user-attachments/assets/567e08cc-b246-4f3d-b7d6-1b4128f274bd)

**Comment by tjanez** [comment]
> Google Photos seems to be broken as well. (version 7.10.0.704893900).

Indeed. I had to downgrade to:
  - Version: 7.9.0.700431741
  - Version code: 49358756
  - https://www.apkmirror.com/apk/google-inc/photos/photos-7-9-0-700431741-release/

For instructions see: https://github.com/microg/GmsCore/issues/2680#issuecomment-2538403448.

**Comment by ale5000-git** [comment]
The problem will likely happen with all Google Apps, no need to report more once the new signature spoofing patch is applied.

**Comment by xundeenergie** [comment]
I know... it's a workaround only... 
But leaving Google behind, you have different solutions:

Use
- Newpipe instead of Youtube-App.
- Use Peertube instead of Youtube
- use Immich instead of Google Photos
- run your own mailserver and use Fairemail or Thunderbird/k9Mail instead of using gmail (Service and app)
- use nextcloud
- use etar, icsx5 davx5 instead of googles calendar and contacts (in connection with nextcloud or other services for contacts and calendar)

and yes... it's only a workaround to this problem. But it's better to leave google behind at all and step into the free and opensource world.

**Comment by phucvu4324** [comment]
> The patch for LineageOS was updated:
> 
> 
This might be stupid question, but can it also can be adapt to a pre-compiled ROMs from a device that abandoned and no longer have any future rom development? My machine doesn't even enough for compile a AOSP ROM from source.

**Comment by ale5000-git** [comment]
> This might be stupid question, but can it also can be adapt to a pre-compiled ROMs from a device that abandoned and no longer have any future rom development? My machine doesn't even enough for compile a AOSP ROM from source.

If Android version is less than 9, no change is needed.
If 9 or higher you need to recompile the ROM yourslef with the change or use [FakeGApps](https://github.com/whew-inc/FakeGApps) (after it is updated with the new signature spoofing changes).

**Comment by ale5000-git** [comment]
The new version of [FakeGApps](https://github.com/whew-inc/FakeGApps) already contains the fix.

**Comment by phucvu4324** [comment]
> The new version of [FakeGApps](https://github.com/whew-inc/FakeGApps) already contains the fix.

thanks!

**Comment by chipironcin** [comment]
The fix for CalyxOS is being tracked here in case you come looking for it:
https://gitlab.com/CalyxOS/calyxos/-/issues/2877

**Comment by DEV-197** [comment]
> > The new version of [FakeGApps](https://github.com/whew-inc/FakeGApps) already contains the fix.
> 
> thanks!

It's still not working for me, same error. Did it work for anyone

**Comment by phucvu4324** [comment]
> > > The new version of [FakeGApps](https://github.com/whew-inc/FakeGApps) already contains the fix.
> > 
> > 
> > thanks!
> 
> It's still not working for me, same error. Did it work for anyone

you need to reboot after enable fakegapps, it working for me with latest youtube app and maps

**Comment by porcoddiocanebastardo** [comment]
Hi, sorry to bother you and thanks a lot for your work!
I still have same problem, changed FakeGApps, rebooted soo many times, different version and way to install microG packages (Revived/MinMicroG/MicroGGApps/direct debloat and installation method), all in different way for two days, no result, I can't update GApps (Map-Calendar-Gmail-Contacts-Messages...) and so I can't use Revanced Youtube packages in mount-root mode has I did before, this time I have to install it in GSMCore way.
Everything work for a year until these Google Updates. I'm a noob but I've always modify Stock Roms of my phones.

In this case I have:
-Ulefone Armor 20WT with Android 12 Stock Rom (build TF3_EEA_V12)
-Root flash boot.img with Magisk
-LSPosed with: FakeGApps6 - GravityBox12.0.0S - DisableFlagSecure2.0.0
-De-Bloater eliminated PlayStore-PlayServices-Framework
-Companion0.3.5 - microG0.3.5 - microGServiceFrameworkProxy all installed without conflicts
-microG 0.3.5 all is ticked OK, only problem in "SafetyNet: CTS profile does not match"

SPIC give errors of pictures and apps too (in picture I try to open Calendar updated version, with old ther's no problems like others GApps).

Thanks a lot for your work and your attention, I'm really so glad to talk and read about you!![Image](https://github.com/user-attachments/assets/271bf611-f218-4efd-909a-471584ac8ae6)
![Image](https://github.com/user-attachments/assets/d8cdb56d-f058-4b77-b339-0d80bf459a23)
![Image](https://github.com/user-attachments/assets/79ecb5b9-bafa-4c11-9d7b-0f135bdc2749)

**Comment by gilbsgilbs** [comment]
@porcoddiocanebastardo You are using Android 12. Did you try the fix I just mentioned: https://github.com/whew-inc/FakeGApps/pull/25 ?

**Comment by ale5000-git** [comment]
Here's a summary of the situation:
- FakeGApps v6.0 or higher ([fix merged](https://github.com/whew-inc/FakeGApps/commit/76891584bcb42eca1001708ee2665ba09c02ebde) and new version already released and working for all Android except 9/10/11/12)
- LineageOS 19.1/20/21/22 ([fix merged](https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/411386))
- /e/OS ([fix merged](https://gitlab.e.foundation/e/os/android_frameworks_base/-/commit/a2b52bdf99f03afe2e36d49fa2b30ffe5bf58a5c))
- crDroid 15 ([fix merged](https://github.com/crdroidandroid/android_frameworks_base/commit/d3dcfccbf266075906f612277ea622a744a3b0bd))
- LineageOS for microG (no info yet)
- DivestOS (no info yet)

**Comment by petefoth** [comment]
> * LineageOS for microG (no info yet)

The code was merged in the upstream  lineage-21.0 branch at 20:52 yesterday. Looking at the [`MicroG LineageOS builds' Matrix room](https://matrix.to/#/#microg-lineage-os-builds:matrix.domainepublic.net) I think that lemonadep should be the first build to have the fix. Other devices will pick up the fix in due course as they are built

- IodéOS: the patches have been applied, and are available in beta updates for some devices. The fix should make it in to the next build run (v5.8)

**Comment by D3SOX** [comment]
Adding to this:
- CalyxOS: Fix merged, included in 6.2.1, currently in Beta (https://calyxos.org/news/2024/12/13/15-december-qpr1-security-update/) and works fine for me

**Comment by ale5000-git** [comment]
> * CalyxOS: Fix merged, included in 6.2.1, currently in Beta (https://calyxos.org/news/2024/12/13/15-december-qpr1-security-update/) and works fine for me

The ticket wasn't updated: https://gitlab.com/CalyxOS/calyxos/-/issues/2877

**Comment by LZeugirdor** [comment]
> Adding to this:
> - CalyxOS: Fix merged, included in 6.2.1, currently in Beta (https://calyxos.org/news/2024/12/13/15-december-qpr1-security-update/) and works fine for me

Once the patch rolls out live will it be safe to update apps or should we stick to older versions still?

**Comment by phucvu4324** [comment]
> > Adding to this:
> > 
> > * CalyxOS: Fix merged, included in 6.2.1, currently in Beta (https://calyxos.org/news/2024/12/13/15-december-qpr1-security-update/) and works fine for me
> 
> Once the patch rolls out live will it be safe to update apps or should we stick to older versions still?

of course, yes

**Comment by grenadin** [comment]
> On the **-hw** version is Huawei that handle the signature spoofing, so only Huawei can change it.
> 
> But are you sure that Huawei is also affected by this issue? Since the way signature spoofing is handled is different the problem may be different or may not happen.

Huawei below P60 was found the problem with this. Above P60 was perfect with 0.3.4 and 0.3.5

**Comment by ale5000-git** [comment]
> Adding to this:
> 
> * CalyxOS: Fix merged, included in 6.2.1, currently in Beta (https://calyxos.org/news/2024/12/13/15-december-qpr1-security-update/) and works fine for me

The fix has been reverted: https://gitlab.com/CalyxOS/platform_frameworks_base/-/commit/4f10cdc88bea8983a4fa7f1a5e63a77fcd1cbca8

What is happened?

**Comment by chirayudesai** [comment]
> The fix has been reverted

Missing null-check, caused issues for some. It has been re-applied and will be rolled out again.

**Comment by ale5000-git** [comment]
> Missing null-check, caused issues for some. It has been re-applied and will be rolled out again.

Could you please link the commit?
I haven't found it here: https://gitlab.com/CalyxOS/platform_frameworks_base/-/commits/staging%2Fandroid15

**Comment by ale5000-git** [comment]
Here's a summary of the situation:
- FakeGApps v6.5 or higher ([fix merged](https://github.com/whew-inc/FakeGApps/commit/76891584bcb42eca1001708ee2665ba09c02ebde), new version released)
- LineageOS 19.1/20/21/22 ([fix merged](https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/411386), all builds from 2024/12/17 should have it)
- LineageOS for microG (fix auto-included from upstream, all builds from 2024/12/17 should have it)
- /e/OS ([fix merged](https://gitlab.e.foundation/e/os/android_frameworks_base/-/commit/a2b52bdf99f03afe2e36d49fa2b30ffe5bf58a5c))
- CalyxOS 6.2.2 and 5.14.1 (in the works)
- crDroid 15 ([fix merged](https://github.com/crdroidandroid/android_frameworks_base/commit/d3dcfccbf266075906f612277ea622a744a3b0bd))
- IodéOS v5.8 or higher (fix included)
- DivestOS (no info yet)

**Comment by pnaf** [comment]
/e/OS 2.6.3 or higher contains the fix: https://community.e.foundation/t/feedback-for-v2-6-3/64267

**Comment by EvanCarroll** [comment]
I see new nightly builds on [download.lineageos.org](https://download.lineageos.org/devices/instantnoodlep/builds). However, they're not on [LineageOS with MicroG](https://download.lineage.microg.org/instantnoodlep/). For those us that did that update and are running the bugged version, will they rebuild LinageOS with MicroG? Can the do a total-rebuild and roll out the fixed version or versioned based on a newer nightly.

It's very inconvenient to have a massively broken version. It seems they're updated the 15th of the month, does this mean without rolling back the next OTA update will be in 9 days?

**Comment by petefoth** [comment]
> I see new nightly builds on [download.lineageos.org](https://download.lineageos.org/devices/instantnoodlep/builds). However, they're not on [LineageOS with MicroG](https://download.lineage.microg.org/instantnoodlep/). For those us that did that update and are running the bugged version, will they rebuild LinageOS with MicroG? Can the do a total-rebuild and roll out the fixed version or versioned based on a newer nightly.

LineageOS for microG L4M and LineageOS (LOS) are ***separate*** projects.  

L4M aim to make builds monthly, and (usually) start a 'build run' starting on the first day of the month. L4M build for the same devices as LOS using [the LOS list of build targets](https://github.com/LineageOS/hudson/blob/main/lineage-build-targets) as the input to our build run.

L4M builds made since LOS merged the necessary code changes around 17th December include the necessary changes, as will the builds currently in progress.

> It seems they're updated the 15th of the month, does this mean without rolling back the next OTA update will be in 9 days?

No. Builds for a device appear when they are made, which will depend on where in the target list they are. The build schedule has changed a bit now that 
1. Lineage 22.1 has launched and 
2. L4M have access to a second build server. See [this post in XDA Forums]https://xdaforums.com/t/lineageos-for-microg.3700997/post-89879147) for more information

You can follow the progress of the L4M builds in the dedicated Matrix room https://matrix.to/#/#microg-lineage-os-builds:matrix.domainepublic.net

**Comment by EvanCarroll** [comment]
> Builds for a device appear when they are made, which will depend on where in the target list they are. 

Ah, so they have can only do one a build month because the build server is bottlenecked? If so, that sucks =( I was hoping that they were just unaware that they needed to rebuild. Does l4m have a dedicated bug tracker?

**Comment by petefoth** [comment]
> Ah, so they have can only do one a build month because the build server is bottlenecked? If so, that sucks =(

If you want to provide us with another build server, don't hold back 😉 

> Does l4m have a dedicated bug tracker?
 Yes: https://github.com/lineageos4microg/docker-lineage-cicd/issues

But please see the ['Troubleshooting and support` secition](https://github.com/lineageos4microg/docker-lineage-cicd?tab=readme-ov-file#troubleshooting-and-support) in our [Docker image repo ~README.md](https://github.com/lineageos4microg/docker-lineage-cicd)

**Comment by brainchild0** [comment]
As a trivial observation, it has apparently not been possible to resolve the issue simply through new releases of the applications related to microg, which users simply would acquire with regular application updates, unrelated to system updates.


Is a design conceivable such that data related to the problem could by patched into the system as a hotfix, without requiring a system upgrade?

For example, could a feature in a user application, or in a system menu, download fixes and then apply them to the deployed system?

Perhaps someone has ideas ideas for a solution through Magisk, not even requiring substantial modifications to microg.

**Comment by ale5000-git** [comment]
> Is a design conceivable such that data related to the issue, or to changes causing similar kinds of breakage, could by patched into the system as a hotfix, without requiring a system upgrade?
> 
> For example, could a feature in a user application, or in a system menu, download fixes and then apply them to the deployed system?

Most likely no.

> Perhaps someone has ideas ideas for a solution through Magisk, not even requiring substantial modifications to microg.

An alternative is to use this Xposed module: [FakeGApps](https://github.com/whew-inc/FakeGApps).

**Comment by brainchild0** [comment]
@ale5000-git: Did users running an official distribution of GApps receive updates to all related components essentially in parallel to the new versions of user applications being published to the official vendor repository, Google Play, or did they also require a system update?

**Comment by ale5000-git** [comment]
Updates to system components aren't related to GApps, but using GApps Google can update some system components.

This issue is about signature spoofing, so completely unrelated to GApps.
Also signature spoofing isn't actually "related" to microG, it is just for other apps to recognize microG as the real Google Play Services.
Essentially microG can works perfectly without signature spoofing as long as the other apps doesn't check its signature.

**Comment by brainchild0** [comment]
@ale5000-git: My general understanding is that the vendor rolled out updates of the user applications, which broke under unofficial environments, but not under the official environment, the combination of AOSP and GApps.

Presumably, users of the latter received updates for various lower and upper layers in tandem, even applying without full system upgrades, that preserved interoperation.

In contrast, resolutions being promoted in the current discussion depend on full system upgrades.

The general sense of my question is how users of environments based in LineageOS or microg might have an experience comparable, in terms of continuity, to the experience generally provided to the vendor.

**Comment by ale5000-git** [comment]
> [@ale5000-git](https://github.com/ale5000-git): My general understanding is that the vendor rolled out updates of the user applications, which broke under unofficial environments, but not under the official environment, the combination of AOSP and GApps.

Yes and no, it is true that it break only in unofficial environments but just by coincidence.
Signature spoofing is used to make microG spoof the signature of official GApps.
GApps is already itself so obviously doesn't need it and it is unaffected :-)

The API used to check signature may return the result in 2 different ways; previously only the first way was used by apps and so previous signature spoofing patches patch only the first way.

Now the Google libraries used by various apps (so inside the various apps) got updated to use the second way of the API (the API is inside the system, so it can't be changed without a full system update).

**Comment by axydavid** [comment]
> > This might be stupid question, but can it also can be adapt to a pre-compiled ROMs from a device that abandoned and no longer have any future rom development? My machine doesn't even enough for compile a AOSP ROM from source.
> 
> If Android version is less than 9, no change is needed. If 9 or higher you need to recompile the ROM yourself with the new patch or use [FakeGApps](https://github.com/whew-inc/FakeGApps).

So i have a question, as far as I see basically we do a small change in ComputerEngine and everything works fine again assuming rom already had signature spoofing.

I have an issue, I'm trying to build crdroid a12.1, now I can find this ComputerEngine in CrDroid a13 version and onwards, but what if you need to update the SignatureSpoofing prior to a13? Where would you look at then?

**Comment by petefoth** [comment]
> I have an issue, I'm trying to build crdroid a12.1, now I can find this ComputerEngine in CrDroid a13 version and onwards, but what if you need to update the SignatureSpoofing prior to a13? Where would you look at then?

The required change for A12 / LOS 19.1 is [here](https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/411385)

For A11 / LOS 18.1 a patch is available, as described in [this XDA Forum post](https://xdaforums.com/t/lineageos-for-microg.3700997/post-89932389). LineageOS for microG used this atch successfully in their recent 'one-off' build run for 18.1 devices, to ge round this issue.

For A10 / LOS 17.1, the changes to AndroidManifest.xml and strings.xml from [this patch](https://github.com/lineageos4microg/docker-lineage-cicd/blob/35b7190bda4f2988845f09e028d266a23fd8b559/src/signature_spoofing_patches/android_frameworks_base-Q.patch). (The other changes in that patch are not needed.) See [this post in the LineageOS for microG issue tracker](https://github.com/lineageos4microg/android_vendor_partner_gms/issues/42#issuecomment-2672597650) for background on this change. And thanks to @Flamefire who is still making up-to-date builds of LineageOS 17.1 for working this out

**Comment by summerlovewarum** [comment]
I found google stealthily installed an APP, Android System Key Verification, which can't be found by manual search in google play store. It was installed without notification of user. After remove this the APPs seems working well. Not sure if this is the root cause of the issue.

**Comment by ale5000-git** [comment]
If you find it again could you please check the internal name of the app?

**Comment by heinhuiz** [comment]
> [@summerlovewarum](https://github.com/summerlovewarum) If you find it again could you please check the internal name of the app?

He is probably referring to the `[Android Aystem Key Verifier](https://developers.google.com/android/binary_transparency/google1p/overview)`.

**Comment by summerlovewarum** [comment]
> > [@summerlovewarum](https://github.com/summerlovewarum) If you find it again could you please check the internal name of the app?
> 
> He is probably referring to the [Android System Key Verifier](https://developers.google.com/android/binary_transparency/google1p/overview).
> 
> [More info here](https://en.androidguias.com/android-system-key-verifier-what-it-is-and-why-it-is-on-your-mobile)

Yes. It's Amdroid System Key Verifier.

**Comment by xtendtech** [comment]
my phone is redmj note 5 and now microg doesn't open. if o click on open it doesn't show anything .I can't link google account to YouTube . anything to try .its android 14

**Comment by vehosah** [comment]
Uhh is the problem got fixed, why is it closed?

**Comment by ale5000-git** [comment]
@vehosah
This problem is already fixed and it is NOT a problem of microG but a problem of the ROM.

**Comment by lintg00** [comment]
@ale5000-git 
How can I fix this problem in Android 11 AOSP?
